### PR TITLE
Improve lookahead docstrings with example and usage notes

### PR DESCRIPTION
**Pull request by rdyro**

### CODE REVIEW

File: optax/_src/lookahead.py [hunk 1]
```diff
@@ -89,6 +89,19 @@ def lookahead(
...
+    >>> import jax.numpy as jnp
+    >>> fast_opt = optax.sgd(1e-2)
+    >>> opt = optax.lookahead(fast_opt, sync_period=5, slow_step_size=0.5)
+    >>> params = optax.LookaheadParams.init_synced(jnp.ones((2,)))
+    >>> state = opt.init(params)
+    >>> # Calculate gradients wrt the fast parameters
+    >>> grads = jax.grad(lambda p: jnp.sum(p**2))(params.fast)
+    >>> updates, state = opt.update(grads, state, params)
+    >>> params = optax.apply_updates(params, updates)
+
```
Comment (vroulet) [inline optax/_src/lookahead.py:108]: Add prints of loss values so that we see that for eval we need the slow params
